contents/blog/the-posthog-array-1-19-0.md

```diff
@@ -0,0 +1,246 @@
+---
+date: 2020-12-15
+title: Array 1.19.0
+rootPage: /blog
+sidebar: Blog
+showTitle: true
+hideAnchor: true
+---
+
+This new release is a great mix between old and new, with significant improvements to both our newer features, as well as our core analytics stack. 
+
+Once again, we were heavily driven by feedback, having done a bunch of calls with our users and actively engaging with the PostHog community on Slack and GitHub. Keep the feedback coming!
+
+## Release Notes
+
+> If you're self-hosting and want to upgrade for a better experience with nicer features, remember to [update your PostHog instance](/docs/configuring-posthog/upgrading-posthog).
+
+### [Scheduled Plugins and Editor](https://github.com/PostHog/posthog/pull/2743)
+
+![Plugin Editor Screenshot](../images/blog/array/plugin-editor.png)
+
+We now support scheduled plugins that run periodically on a specified time cycle (e.g. minute, hour, day), as well as have a built-in code editor for plugins right into the PostHog UI.
+
+With the ability to run tasks in specified time intervals, you can now setup plugins that, for example, keep track of external metrics and add this data to PostHog via new events. This is possible because we now [support `posthog.capture` calls inside plugins as well](https://github.com/PostHog/posthog-plugin-server/pull/67). 
+
+Some metrics you might want to keep track of are, for example, server performance, GitHub activities (e.g. stars ⭐ ), engagement with your project's social media profiles, and anything else you can think of!
+
+Here's an example to give you an idea:
+
+```js
+async function runEveryMinute({ config }) {
+    const url = `https://api.github.com/repos/PostHog/posthog`
+    const response = await fetch(url)
+    const metrics = await response.json()
+
+    posthog.capture('github metrics', {
+        stars: metrics.stargazers_count,
+        open_issues: metrics.open_issues_count,
+        forks: metrics.forks_count,
+        subscribers: metrics.subscribers_count
+    })
+}
+```
+
+You can learn more about scheduled plugins on the [PR that created them](https://github.com/PostHog/posthog-plugin-server/pull/63), as well as our docs for [building your own plugin](/docs/plugins/build).
+
+> **Note:** Plugins are a Beta feature currently only available on self-hosted instances. We are working to make it available on PostHog Cloud soon.
+
+### [Lifecycle Analysis](https://github.com/PostHog/posthog/pull/2460)
+
+![Lifecycle Screenshot](../images/blog/array/lifecycle.png)
+
+Our 'Trends' tab just got an awesome new feature: lifecycle graphs!
+
+Lifecycle analysis digs deeper into your events and shows you a breakdown of the event volume into new, returning, and resurrecting occurrences. In addition, it also shows you the churn on that event for the specified time period. 
+
+To use it, select 'Shown As' -> 'Lifecycle' when in the 'Trends' tab.  
+
+### [Retention Graph](https://github.com/PostHog/posthog/pull/2578)
```

Comment:
I would remove this showcase for now. The cohorts graph is quite useful when you have all the cohorts graphs in there, but if you only have a single cohort is not really that helpful. There's a discussion for this on the relevant issue, we're adding this soon.

Reply:
Remove the highlight altogether or the pic you mean?

It's quite a nice view of retention over time, no? User today during the call was quite happy to see it

Reply:
I would remove the highlight altogether, but just a suggestion. It doesn't really show retention over time, it shows the retention of a _specific cohort_ over time (not clear which cohort). I think it can be confusing and not very standard when doing product analytics, that's why I would remove it, but not pushing very strongly on this.

Reply:
got it, will leave for a release later down the line!

thanks for the context